core/include/macros/math.h

```diff
@@ -31,7 +31,7 @@ extern "C" {
                               unsigned int: 1,           \
                               unsigned long: 1,          \
                               unsigned long long: 1,     \
-                              default: (long long)(a) < 0 ? -1 : 1)
+                              default: ((a) > 0LL) ? 1 : -1)
```

Comment:
```suggestion
                              default: ((a) <= 0 ? ((a) == 0 ? 0L : -1L ): 1L))
```

for this variant the compile no longer complains, and if it is build the result isn't to bad

Reply:
or evne better this one:
 it will provide the user with the same result as the original ( -inf .. 0[ == -1 and [0 .. inf ==1 ) in two comparisons (if not optimized) so the warning is happy and when it is optimized it has the same result
```suggestion
                              default: ((a) <= 0 ? ((a) == 0 ? 1L : -1L ): 1L))
```
see:
https://godbolt.org/z/nE97rKzPe